zerver/decorator.py

```diff
@@ -751,10 +751,12 @@ def _wrapped_func_arguments(request: HttpRequest, *args: Any, **kwargs: Any) ->
     return _wrapped_view_func
 
 # Converter functions for use with has_request_variables
-def to_non_negative_int(s: str) -> int:
+def to_non_negative_int(s: str, max_int_size: int=None) -> int:
     x = int(s)
     if x < 0:
         raise ValueError("argument is negative")
+    if max_int_size and x > max_int_size:
+        raise ValueError('%s is too large (max %s)' % (x, max_int_size))
```

Comment:
This should come with tests in `test_validator.py`.  We might also want to move this function to `zerver/lib/validators.py`